service/src/main/java/uk/gov/hmcts/reform/fpl/service/respondent/RespondentAfterSubmissionValidator.java

```diff
@@ -52,18 +66,45 @@
             if (YES.getValue().equals(previous.getLegalRepresentation())
                 && NO.getValue().equals(current.getLegalRepresentation())) {
                 errors.add(String.format("You cannot remove respondent %d's legal representative", i + 1));
+                continue;
             }
 
             if (getLegalRepresentation(current).equals(getLegalRepresentation(previous))
                 && !Objects.equals(getOrganisationID(current), getOrganisationID(previous))) {
 
-                errors.add(String.format("Change of organisation for respondent %d is not allowed", i + 1));
+                errors.add(String.format(
+                    "You cannot change organisation details for respondent %d's legal representative", i + 1));
+                continue;
             }
+
+            errors.addAll(legalRepresentationErrors(current, i));
         }
 
         return errors;
     }
 
+    private List<String> legalRepresentationErrors(Respondent respondent, int i) {
+        List<String> errors = new ArrayList<>();
+        if (respondent.getLegalRepresentation() == null) {
+            errors.add(String.format("Confirm if respondent %d has legal representation", i + 1));
+        } else if (YES.getValue().equals(respondent.getLegalRepresentation())) {
+            {
+                if (isEmpty(respondent.getSolicitor().getFirstName())
```

Comment:
Would also be sensible just passing `i+1` as the parameter as you aren't using `i` in any other way